components/funnel.js

```diff
@@ -0,0 +1,103 @@
+// funnel.js
+
+var _ = require('underscore');
+var fs = require('fs');
+
+var stateFactory = require('../src/create-state.js');
+var logger = require('../src/logger.js');
+var wrapper = require('../src/javascript-wrapper.js');
+
+module.exports = wrapper(funnel);
+
+/**
+ * Funnels a single input into the pipeline one at a time, and queues the rest
+ * Fires the next one when it receives the id that has just finished executing.
+ *
+ * @param input Either an id, to be queued and sent to the output when it is its turn.
+ *              The input value will be stored as a patientId attribute in the VNI 
+ *              metadata, which will be passed to all nodes down stream.  The input 
+ *              port is NOT multi - to work properly, it is expected to have two inputs: 
+ *                 - the input edge that feeds in one or more id values
+ *                 - an input edge that returns the id value when it has finished 
+ *                   downstream processing and the next id should be sent through the funnel.
+ *              This code will need to be changed if the pipeline is modified to disallow multi.
+ *
+ * @param metadata_key a key name that will be used to create metadata on the VNI to send downstream.
+ *                     In the LPI RDF Pipeline, this would be patientId, but it could be any 
+ *                     name the pipeline builder wishes to use.  If it is empty, the key name 
+ *                     funnelId will be used.
+ */
+function funnel(input, metadata_key) {
+
+    var node = this.nodeInstance;
+    var metadataKey = _.isEmpty(metadata_key) ? 'funnelId' : metadata_key;
+    if (!_.isString(metadataKey)) throw Error('Funnel requires a metadata key string!'); 
+
+    if (_.isString(input)) 
+        console.log('\nenter funnel with input:',input);
+    else 
+        console.log('\nenter funnel with non-string input');
+
+    // Initialize the funnel state in node instance the first time through
+    if (_.isUndefined(node.funnel)) {
+        node['funnel'] = {};
+        node.funnel.executed = 0;
+        node.funnel.queue = []; 
+        node.funnel.executing = '';
+    }
+
+    var funnel = node.funnel;
+    var funnelInput = input;
+
+    // Input coming over http will have extraneous leading and trailing quotes embedded - remove 'em
+    if (!_.isEmpty(funnelInput)) {
+        if (_.isString(funnelInput) && funnelInput.charAt(0) === '"' && funnelInput.charAt(input.length -1) === '"') 
+            funnelInput = funnelInput.substr(1, funnelInput.length -2);
```

Comment:
It does not see this as JSON because it is coming in as a single field e.g., "2-000007" .  The JSON parser expects a key value pair.  I tried feeding it values I am getting from http-accept -server, like those above, and the JSON parser threw an exception. 

If you have a key value pair and are stripping off just the value, then yes, that would be helpful.  The stripping first and last quote if they are present is a hack, but it was a necessary one with the data I was getting.